index.qmd

```diff
@@ -0,0 +1,65 @@
+---
+title: "Arrow New Feature Showcase: `show_exec_plan()` "
+format: html
+editor: visual
+---
+
+The `arrow` package allows you to take advantage of the power of the [Acero execution engine](https://arrow.apache.org/docs/cpp/streaming_execution.html) for data manipulation and analysis. The code in `arrow` provides bindings to `dplyr` verbs and `tidyverse` functions, so that you can use these interfaces without having to understand the inner workings of Acero. But what if you actually want to know more?
+
+In the latest release of `arrow`, version 9.0.0, the [`show_exec_plan()`](https://arrow.apache.org/docs/r/reference/show_exec_plan.html) function is introduced. This function allows you to see the execution plan generated from your code. For example:
+
+```{r}
+#| echo: false
+#| output: false
+library(arrow)
+library(dplyr)
+```
+
+```{r}
+library(arrow)
+library(dplyr)
+mtcars %>%
+  arrow_table() %>%
+  filter(mpg > 20) %>%
+  mutate(wt_kg = (wt*1000) * 0.453592)  %>%
+  show_exec_plan()
+```
+
+This functionality is similar to that of [`dplyr::explain()`](https://dplyr.tidyverse.org/reference/explain.html) and you'll get the same result whether you call [`arrow::show_exec_plan()`](https://arrow.apache.org/docs/r/reference/show_exec_plan.html), [`dplyr::explain()`](https://dplyr.tidyverse.org/reference/explain.html) or [`dplyr::show_query()`](https://dplyr.tidyverse.org/reference/explain.html) on an `arrow_dplyr_query` object.
+
+## What is Acero?
+
+Acero is Arrow's query execution engine.
+
+## What is a query execution engine?
+
+A query execution engine is a piece of software which allows users to execute queries against a data source. It typically has multiple steps, which could include things like taking the query and parsing it into an algebraic format, re-ordering and optimising the query in order to run it in the most efficient way, and actually running the query.
+
+## What is an ExecPlan?
```

Comment:
I like your explanation here. One thing worth mentioning is that unlike other query engines like DuckDB, Acero has no optimizer. So it won't change the order to steps to be optimal for you; you have to do this yourself.

Two possible ways you could expand on this:

 (1) Provide plain-language definitions of each of the exec node types (you have a start, but providing a complete list is nice).
 (2) Show and example where reordering the query makes it faster (like switching a filter and project around)

Reply:
Thanks for those suggestions!

I'm not entirely sure about what is or is not in scope for this blog post, but this is stuff I'm interesting in learning and clarifying my own knowledge of anyway, so I'll get it written, and if it ends up being better located elsewhere, then that's just more documentation written for the project so it's a win regardless.